frontend/src/players/players-list/players-list.tsx

```diff
@@ -96,43 +95,38 @@ export const PlayersList = ({ team, onFormSubmit }: Props): JSX.Element => {
   }, [team.players]);
 
   const deletePlayer = (playerId: number) => {
-    api.deletePlayer(team.id, playerId);
-    setPlayers(players.filter((player) => player.id !== playerId));
+    api.deletePlayer(team.id, playerId).then(() => {
+      setPlayers(players.filter((player) => player.id !== playerId));
```

Comment:
You should not map the behaviour of the backend to the frontend. You also don't need the state, `Players` are available in the `team` props. When you remove the `Players` there should be a re-download of the `team` which will reload the components.